test/allowlist_for_publicAPI.json

```diff
@@ -694,6 +694,21 @@
     "kineto_available",
     "record_function"
   ],
+  "torch.pytree": [
+    "register_node",
+    "all",
+    "all_only",
+    "any",
+    "any_only",
+    "flatten",
+    "iter",
+    "leaves",
+    "map",
+    "map_",
+    "map_only",
+    "map_only_",
+    "structure"
+  ],
```

Comment:
> New entries in the allowlist for public API are never ok. Please make sure public APIs are properly done.

@albanD These functions are already exported in `torch.utils.pytree`. The `__module__` attribute of these functions is `torch.utils.pytree`. We cannot change it to `torch.pytree` because we will add `torch.utils.pytree` to the allowlist instead.

Reply:
Re-exposing at a higher level is valid from the public apis spec point of view. So this should not require you to put things here.

Reply:
If it's already public API in torch.utils.pytree, why do you need to re-expose it higher?
I think it's fair to keep it there.

Reply:
> If it's already public API in torch.utils.pytree, why do you need to re-expose it higher?

The rational is listed in https://github.com/pytorch/pytorch/pull/148180#issue-2886832964. FYI, the pytree functions in JAX are also exported in two places: `jax.tree` and `jax.tree_util`.